src/main/java/core/basesyntax/dao/impl/CommentDaoImpl.java

```diff
@@ -1,32 +1,72 @@
 package core.basesyntax.dao.impl;
 
 import core.basesyntax.dao.CommentDao;
+import core.basesyntax.exception.DataProcessingException;
 import core.basesyntax.model.Comment;
 import java.util.List;
+import org.hibernate.Session;
 import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
 
 public class CommentDaoImpl extends AbstractDao implements CommentDao {
     public CommentDaoImpl(SessionFactory sessionFactory) {
         super(sessionFactory);
     }
 
     @Override
-    public Comment create(Comment entity) {
-        return null;
+    public Comment create(Comment comment) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(comment);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't insert Content entity", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return comment;
     }
 
     @Override
     public Comment get(Long id) {
-        return null;
+        try (Session session = factory.openSession()) {
+            return session.get(Comment.class, id);
+        } catch (Exception e) {
+            throw new DataProcessingException("can't get Comment", e);
+        }
     }
 
     @Override
     public List<Comment> getAll() {
-        return null;
+        try (Session session = factory.openSession()) {
+            Query<Comment> getAllCommentQuery = session.createQuery("from Comment ", Comment.class);
+            return getAllCommentQuery.getResultList();
+        } catch (Exception e) {
+            throw new DataProcessingException("Not able to retrieve all comments from database", e);
+        }
     }
 
     @Override
-    public void remove(Comment entity) {
-
+    public void remove(Comment comment) {
+        Transaction transaction = null;
+        try (Session session = factory.openSession()) {
+            transaction = session.beginTransaction();
+            session.remove(comment);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Comment was not removed", e);
```

Comment:
you won't be able to rollback the transaction because the session where you began the transaction - is already closed (because of try-with-resources)